packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection/PersonalDetailsForm.tsx

```diff
@@ -85,11 +86,15 @@ export const PersonalDetailsForm = () => {
   function onSubmit(
     userDetails: PersonalDetailsFormFields,
   ): SubmitHandler<PersonalDetailsFormFields> {
-    const updates: UserAccountDetails = Object.fromEntries(
-      Object.entries(userDetails).filter(([field]) => dirtyFields[field]),
-    );
-
-    updateUser(updates);
+    console.log(userDetails);
+    const updates: PersonalDetailsFormFields = Object.keys(dirtyFields)
```

Comment:
I'm a bit late to the review party but I just wanted to chime in and say that I support this extra validation but I reckon it could be made a little bit drier somehow. Some options could be:
- make a re-usable base TextField Component
- extend react-hook form [eg](https://github.com/orgs/react-hook-form/discussions/9354)
- do the trimming on submit handling [eg](https://github.com/react-hook-form/react-hook-form/issues/1650)

I don't feel too strongly about it - just an optional suggestion

Reply:
I agree that readability is lacking here, but in this case I’m don’t quite understand what you mean by calling with relevant data. Do you have an idea/example of how you would call this helper function?

In the meantime, I have been able to tidy up the logic for **validating** that the form fields aren’t just whitespace chars, but the bit where I trim them for submission is still maybe a bit too clever and hard to read

Reply:
Sorry, I mean call it with the form data

Reply:
I found that breaking it into `keepOnlyModifiedFields()` and `trimWhitespaces()` wasn’t really more readable than two short inline comments, so have just done the latter